pkg/routeagent_driver/handlers/ovn/gateway_route_handler.go

```diff
@@ -68,20 +68,28 @@ func (h *GatewayRouteHandler) GetNetworkPlugins() []string {
 
 func (h *GatewayRouteHandler) RemoteEndpointCreated(endpoint *submarinerv1.Endpoint) error {
 	if h.State().IsOnGateway() {
-		_, err := h.smClient.SubmarinerV1().GatewayRoutes(endpoint.Namespace).Create(context.TODO(),
-			h.newGatewayRoute(endpoint), metav1.CreateOptions{})
-		if err != nil && !apierrors.IsAlreadyExists(err) {
+		logger.V(log.TRACE).Infof("Remote endpoint %q created event received on gateway node", endpoint.Name)
+
+		gwr := h.newGatewayRoute(endpoint)
+
+		result, err := util.CreateOrUpdate(context.TODO(), GatewayResourceInterface(h.smClient, endpoint.Namespace),
+			gwr, util.Replace(gwr))
+		if err != nil {
 			return errors.Wrapf(err, "error processing the remote endpoint creation for %q", endpoint.Name)
 		}
+
+		logger.V(log.TRACE).Infof("GatewayRoute %s: %#v", result, gwr)
```

Comment:
Instead of logging the entire gwr, can we simply log the gwr.RoutePolicySpec alone?

Reply:
The only other field would be the name which may also be useful to see. Also use the more readable `resource.ToJSON`. And I think we can log this as just Info - it shouldn't cause excessive logging. Also we can combine L71 above.

```suggestion
		logger.Infof("GatewayRoute %s from remote endpoint %s: %s", result, endpoint.Name, resource.ToJSON(gwr))
```

Reply:
changed to json and removed trace